hoomd/ParticleGroup.cc

```diff
@@ -789,6 +792,100 @@ unsigned int ParticleGroup::intersectionSize(std::shared_ptr<ParticleGroup> othe
     return n;
     }
 
+void ParticleGroup::thermalizeParticleMomenta(Scalar kT, unsigned int seed, unsigned int timestep)
+    {
+    unsigned int group_size = this->getNumMembers();
+
+    const unsigned int D = Scalar(m_sysdef->getNDimensions());
+
+    ArrayHandle<Scalar4> h_vel(m_pdata->getVelocities(),
+                               access_location::host,
+                               access_mode::readwrite);
+
+    ArrayHandle<Scalar4> h_orientation(m_pdata->getOrientationArray(),
+                                       access_location::host,
+                                       access_mode::read);
+
+    ArrayHandle<Scalar4> h_angmom(m_pdata->getAngularMomentumArray(),
+                                  access_location::host,
+                                  access_mode::readwrite);
+
+    ArrayHandle<Scalar3> h_inertia(m_pdata->getMomentsOfInertiaArray(),
+                                   access_location::host,
+                                   access_mode::read);
+
+    ArrayHandle<unsigned int> h_tag(m_pdata->getTags(),
+                                    access_location::host,
+                                    access_mode::read);
+
+    // Total the system's linear momentum
+    vec3<Scalar> tot_momentum(0, 0, 0);
+
+    // Loop over all particles in the group
+    for (unsigned int group_idx = 0; group_idx < group_size; group_idx++)
+        {
+        unsigned int j = this->getMemberIndex(group_idx);
+        unsigned int ptag = h_tag.data[j];
+
+        // Seed the RNG
+        hoomd::RandomGenerator rng(hoomd::RNGIdentifier::ParticleGroupThermalize, seed, ptag, timestep);
+
+        // Generate a random velocity
+        Scalar mass =  h_vel.data[j].w;
+        Scalar sigma = fast::sqrt(kT / mass);
+        hoomd::NormalDistribution<Scalar> normal(sigma);
+        h_vel.data[j].x = normal(rng);
+        h_vel.data[j].y = normal(rng);
+        if (D > 2)
+            h_vel.data[j].z = normal(rng);
+        else
+            h_vel.data[j].z = 0; // For 2D systems
```

Comment:
I'm personally a fan of the ternary operator for such short snippets, especially when you're choosing not to enclose the if/else blocks in braces because they're so short (that's a clear sign for me that `? :` syntax is appropriate. Not sure if you have are stylistically opposed, though.

Reply:
I'm not a fan of the ternary operator. I'll run clang-format on this function and see what the style rules we have set do with these blocks.

Reply:
clang-format leaves these simple if's as is with the current configuration. We can discuss at a later date what the style settings should be before we run clang-format on the code base.

Reply:
OK sounds good.